lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp

```diff
@@ -265,3 +277,107 @@ std::unique_ptr<OperationPass<ModuleOp>>
 mlir::torch::Torch::createVerifyBackendContractPass() {
   return std::make_unique<VerifyBackendContractPass>();
 }
+
+static void markDecomposedOpsAsIllegal(MLIRContext *context,
```

Comment:
Is there an appropriate place where you can put a lightweight "architecture" comment describing the overall approach here (since it crosses multiple files)?

Since this change actually formalizes the backend contract better (now satisfiesBackendContract will check the legal set of ops, rather than having that "satisfaction" check hidden inside DecomposeComplexOps), we probably want a nice big comment in this file for when folks read "See the extensive comments in the function satisfiesBackendContract (and its callees) in the LowerToBackendContract pass for an extended rationale for these decisions, and a precise definition of the backend contract." in https://github.com/llvm/torch-mlir/blob/main/docs/architecture.md#the-backend-contract

Reply:
I've added a comment explaining the rationale. Let me know if you want me to change anything